### PR TITLE
Allow defining error codes ignored by the cache

### LINKED ISSUE #56: antivirus script handling: differentiate between infected and av not available
Currently when the antivirus is not available the file is rejected as infected, which is the good thing to do.

However this result is also cached, which gives us false positive even after the antivirus became available since the result is taken from the cache.

We should use a specific return code for when the antivirus is not avail, and not cache the result in this case.

### PR DESCRIPTION
**Pull request by babolivier**
This is to avoid caching an accidental failure due to the AV being down (e.g. restarting etc).

Will require https://github.com/matrix-org/matrix-dinsic/pull/829 to be merged and deployed before being usable on DINUM's infra.

Fixes #56

### CODE REVIEW

**Review by squahtx — APPROVED** [review]
lgtm

**Review by squahtx — APPROVED** [review]
lgtm
however let's get someone else to take a quick look in case I've missed something again

File: src/reporting.js [hunk 1]
```diff
@@ -366,7 +366,14 @@ async function generateReport(console, httpUrl, matrixFile, filePath, tempDir, s
     console.info(`Running command ${cmd}`);
     const result = await executeCommand(cmd);
 
-    reportCache[reportHash] = result;
+    const config = getConfig();
+
+    if (
+        !config.scan.doNotCacheExitCodes
```
Comment (reivilibre) [inline src/reporting.js:372]: In the interest of being paranoid around JS, I think I would prefer to make this explicit.
What are you interested in here? Non-null/undefinedness? Emptiness?

This is apparently not a very Pythonic land:
```
> ![]
false
> [] == true
false
> [] == false
true
```

File: test/reporting-test.js [hunk 2]
```diff
@@ -134,5 +144,36 @@ describe('reporting.js', () => {
 
             assert.strictEqual(report.clean, false);
         });
+
+        it('should not cache if a scan failed with an exit code that should be ignored', async () => {
```
Comment (reivilibre) [inline test/reporting-test.js:148]: any chance we can test that it DOES cache in the opposite case?

**Review by reivilibre — CHANGES_REQUESTED** [review]

File: src/reporting.js (hunk 1 above)
Comment (babolivier) [reply src/reporting.js:372]: I'm interested in all kind of false, i.e. I want this line to be `true` unless `config.scan.doNotCacheExitCodes` is a non-empty array, but I guess I can do these two checks separately.

File: test/reporting-test.js (hunk 2 above)
Comment (babolivier) [reply test/reporting-test.js:148]: Sure

File: src/reporting.js (hunk 1 above)
Comment (reivilibre) [reply src/reporting.js]: thanks, I think it's much more sane to have the code spell it out clearly as it does now (especially for non-JSers)

File: test/reporting-test.js (hunk 2 above)
Comment (reivilibre) [reply test/reporting-test.js:148]: fab, it always feels like we're only telling half the story if we only test when it doesn't happen :).

**Review by reivilibre — APPROVED** [review]
Thanks!

**Pull request merged**